### ISSUE TITLE
Search component

### ISSUE THREAD

**Issue by ddelpiano** [MetaCell/geppetto-NeuroSCAN#43]
UI:

- component styling (done)
- implement the input text (done)
- implement a loading spinner blocking the search component while waiting on response (done)
- implement the slider that represents the developmental stages (done)
- implement the expandable sections for the results structured as (done)
  - neurons
  - contacts
  - synapses
- Implement splitted button to instantiate new cphate viewers (done)
- hovering one results record should highlight that and show the button "add to" (done)
- implement the "add to" popup/menu (done)
- implement the scrolling when more results are loaded

Query and redux:

- implement the "load more" button in the results, a click on load more will send another request to fetch entries starting from the limit of the last result (Strapi CMS provides `start` and `limit` parameters to implement this behavior, all endpoints come with the same API features listed in [their documentation](https://strapi.io/documentation/developer-docs/latest/developer-resources/content-api/content-api.html#limit)). Since the number of neurons is limited to 200-300, we can fetch these in one request by setting a high limit parameter. However, we still might want to show only the first 10-30 results even if they are already present. Therefore, the click more mechanism will consider (1) number of items to display and (2) available items returned in requests.
- Connect dev stage slider to the search query.
- Connect redux store and link redux action to pop up menu.
- Connect "Add CPHATE button" to the dev time slider, when a new cphate is added to the scene the timepoint chosen should be the one selected in the search component at the moment of the click.
- The results count for each category (neurons/synapses/contacts) should account for all the instances that matches the search, so we run a count query first and then we fetch when needed (or both at the same time).
- Synapses search can filter for a single neuron in the pre side.
- Synapses search can filter for a single neuron in the post side.

**Functional requirements for the search:**

**Neurons:**

- for each string/chip given in input we display the neuron(s) that match the input given
- if the user put in input 2 strings/chips we display both neurons that are matching these strings.
- The results are given in a flat list.

**Contacts:**

- The results are given in form of a flat list of contacts in the shape of "NeuronA contact NeuronB"
search all contacts where the neuron in input is either NeuronA or NeuronB
- If the user gives in input 2 strings/chips (ADAL and AIBR), the results will be all the contacts that have these 2 strings, bidirectionally. So, assuming the data exists, we will have a flat list with:
ADAL-AIBR
AIBR-ADAL
- If the user inputs 3 strings/chips we will look for one occurrence that has the all 3 of them, we will NOT work on permutations of all the chips given.

**Synapses:**

- the results will be given in form of a flat list of synapses in the shape of pre-NeuronA-type-post-NeuronB
- we'll search all synapses where the input is part of the pre or of the post connection
the filter will show checkboxes with the 2 types chemical/eletrical (see image attached).
- If the user inputs 3 strings/chips we will look for one occurrence that has the all 3 of them, we will NOT work on permutations of all the chips given.
- We will find all the synapses where the 2/3/4 neurons in input are part of the pre and post connection, so bidirectionally.

**Comment by ddelpiano** [comment]
moved to icebox since I think it would be better to split this into smaller tasks and see what we can fit in the release 0.1

**Comment by ddelpiano** [comment]
@lrebscher decreasing this estimate by the hours allocated for https://app.zenhub.com/workspaces/yale-neuroscan--db-promoter-60749ce5c09a74000fe7ef74/issues/metacell/geppetto-neuroscan/122 so we keep in line all the numbers to make them match.

**Comment by zsinnema** [comment]
we can use this for the filtering: https://github.com/ljharb/qs
